lib/js-preprocessor/broccoli-generate-components.js

```diff
@@ -0,0 +1,73 @@
+/* jshint node: true */
+'use strict';
+
+var fs = require('fs');
+var path = require('path');
+
+var CachingWriter = require('broccoli-caching-writer');
+var walkSync = require('walk-sync');
+var mkdirp = require('mkdirp');
+
+GenerateComponents.prototype = Object.create(CachingWriter.prototype);
+GenerateComponents.prototype.constructor = GenerateComponents;
+
+function GenerateComponents(inputTree, options) {
+  CachingWriter.call(this, [inputTree]);
+
+  this.registryType = options.registryType;
+  this.moduleName = options.moduleName;
+  this.prefix = options.prefix;
+  this.baseDir = options.baseDir;
+  this.scss = options.scss;
+  this.js = options.js;
+}
+
+GenerateComponents.prototype.build = function() {
+  var components = componentsToGenerate(this.baseDir, this.scss, this.js);
+
+  components.forEach(function(hash) {
+    this.generateFile(hash.component, hash.default);
+  }, this);
+};
+
+GenerateComponents.prototype.generateFile = function(fileName, defaultFile) {
+  var exportFrom = this.registryType === 'self' ? defaultFile : fileName;
+  var module = exportFrom.replace(/(.+)\.js$/, this.moduleName + '/$1');
+  var component = "export { default } from '" + module + "';\n";
+
+  var outputFile = path.join(this.outputPath, this.prefix, fileName);
+  var outputDir = path.dirname(outputFile);
+
+  mkdirp.sync(outputDir);
+  fs.writeFileSync(outputFile, component, 'utf8');
+};
+
+var componentsToGenerate = function(baseDir, scss, js) {
+  var scssFiles = walkSync(baseDir, { globs: scss });
+  var jsFiles = walkSync(baseDir, { globs: js });
+
+  var expectedJsFiles = scssFiles.map(function(fileName) {
+    return fileName.replace(/^styles\/components\/(.+)\.scss/, 'components/$1.js');
```

Comment:
Thanks for the clarification. This is the place to filter those out for sure. I'll have to remove the tests as well since those deal with base components.

- [ ] filter out base components